### DISCUSSION TITLE
When clicking on the container, determine which page to redirect the user to

### DISCUSSION THREAD

**Discussion by Meksis** [flet-dev/flet#2006] [Q&A]
### Question

Hello. I'm making something like a small store catalog. From the SQL database I get information about what products are currently in the store, after which I go through them in a loop and add them to the GridView in the form of a clickable container with pictures. When I click on a container, I want it to redirect me to that product's page, but no matter what I do, it always ends up that I only get a link to the very last item in the store - the rest are simply skipped. I can't figure out what's wrong. If anyone knows, help) A piece of code is below.

### Code sample

```python
import flet as ft
from Utils.DB_work_sync import DB_ORM



def Main(page):
    page.title = 'Каталог'
    DB_object = DB_ORM()
    DB_object.connect()

    # page.update()


    def main():



        items_on_market = DB_object.execute("""SELECT * FROM ITEMS""", is_change = False)

        items = ft.GridView(expand=True,
                            runs_count=5,
                            max_extent=100,
                            padding = 10,
                            child_aspect_ratio=1.0,
                            spacing=10,
                            run_spacing=8)


        items.controls = [ft.Container(ft.Text(item[1]), on_click=lambda x: print(f'Pressed container with id = {item[1]}')) for item in items_on_market]



        page.controls.append(items)



        page.update()
    
    main()




ft.app(target=Main, view=ft.AppView.WEB_BROWSER)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ingdesarr1** [comment]
Can you pass a example of items_on_market to validate? Thanks

**Reply by Meksis** [reply]
Sure!

![изображение](https://github.com/flet-dev/flet/assets/107752360/c6636342-a7ff-4665-a610-ff6a2ab70b60)

**Reply by ingdesarr1** [reply]
items.controls.append(ft.Container(ft.Text(item[1]), on_click=lambda x: print(f'Pressed container with id = {item[1]}')) for item in items_on_market)

if you try in this way, work?

**Reply by Meksis** [reply]
No, you just appending controls list but these also stay a similar objects

**Comment by ndonkoHenri** [comment]
Add a `data` property to your container giving it the item it should store, then from the `lambda`, instead print `e.control.data`

**Reply by Meksis** [reply]
I tried it, but when the Container gets clicked, it returns a ClickEvent that doesn't contain any data provided to the container.

**Reply by ndonkoHenri** [reply]
Can you please paste the code you use ?
